pkgs/pkgs-lib/formats.nix

```diff
@@ -34,7 +39,7 @@ rec {
   inherit (import ./formats/java-properties/default.nix { inherit lib pkgs; })
     javaProperties;
 
-  json = {}: {
+  json = { ... }: {
```

Comment:
Why is `...` added to all the formatters? As far as I can see this is not needed

Reply:
From my understanding, the `...` was added so that all of the formatters accept the same arguments (even if unused) ,so that you could treat the formatter as arbitrary. (Any arguments they don't use are just ignored, so you could swap out without issue.)

Reply:
I thought it made sense as some of the formatters have `...` (ini, keyValue, gitIni), but not all (javaProperties, elixirconf, toml, yaml, json) so perhaps all of them should (for consistency).
It's so that they can be used generically, ignoring the arguments they don't support instead of erroring out. But I can not do this if it makes more sense.

Reply:
Originally I added `...` only to the `ini` formatter because the `generators.toINI` supported extra arguments that I didn't want to all duplicate.


> so that you could treat the formatter as arbitrary. (Any arguments they don't use are just ignored, so you could swap out without issue.)

I think it's better to get an error in such a case, it protects against spelling mistakes, e.g. `yaml { coment = "nope"; }` should error, not just ignore the argument. And this makes it clear which arguments are supported by which functions. The formats are individual and have very individual arguments, they don't need to share the same ones.